docs/001_develop/03_client-capabilities/023_custom-components/05_directives.mdx

```diff
@@ -2,114 +2,146 @@
 
-The ref directive lets you directly reference a specific DOM element within your component's template, enabling direct interaction with that element.
+The `ref` directive lets you directly reference a specific DOM element within your component's template, enabling direct interaction with that element.
 
-Here’s how to use ref in `<my-button>` to reference the button element, making it accessible for direct manipulation in code.
+Here’s how to use `ref` in `<my-button>` to reference the button element, making it accessible for direct manipulation in code.
 
 ```typescript
-import {customElement, GenesisElement, html, ref } from '@genesislcap/web-core';
+import {
+  customElement,
+  GenesisElement,
+  html,
+  ref,
+} from "@genesislcap/web-core";
 
 @customElement({
-    name: "my-button",
-    template: html`
-        <button ${ref("buttonElement")}>Click Me</button>
-    `,
+  name: "my-button",
+  template: html<MyButton>` <button ${ref("buttonElement")}>Click Me</button> `,
 })
 export class MyButton extends GenesisElement {
-    buttonElement: HTMLButtonElement;
+  buttonElement: HTMLButtonElement;
 
-    connectedCallback() {
-        super.connectedCallback();
-        console.log("Button text:", this.buttonElement.textContent);
-    }
+  connectedCallback() {
+    super.connectedCallback();
+    console.log("Button text:", this.buttonElement.textContent); // Button text: Click Me
+  }
 }
-
 ```
 
-- `ref("buttonElement")`: The ref directive assigns the `<button>` element to the `buttonElement` property on the `<my-button>` component instance.
+##### Key Points:
 
-- In `connectedCallback`, `this.buttonElement` is a direct reference to the `<button>` DOM node, allowing us to access properties like `textContent`. Here, we log the button’s text to the console.
+>- `ref("buttonElement")`: The `ref` directive creates a reference to the `<button>` element and assigns it to the `buttonElement` property of `<my-button>` component instance, allowing direct interaction with the DOM element.
+
+>- In `connectedCallback`, `this.buttonElement` is a direct reference to the `<button>` DOM node, allowing us to access properties like `textContent`. Here, we log the button’s text to the console.
 
 :::note
 
 Even though the `<button>` element is part of our `<my-button>` component’s template, you can’t directly access DOM elements inside the template without a reference. The `ref` directive provides this direct access, allowing you to manipulate or retrieve information from a specific DOM element.
 
 :::
 
-### Syntax breakdown
+#### Syntax breakdown
 
 ```typescript
 ref(propertyName);
 ```
-- Property Name: The name of the property that will store the DOM reference in the component. The element with ref(propertyName) will be assigned to this property.
 
-## Slotted
+- Property Name: The name of the property that will store the DOM reference in the component. The element with `ref(propertyName)` will be assigned to this property.
+
+### Slotted
```

Comment:
Rereading this has made me remember what my previous comment was that we were confused by. I think that it isn't great that the first time we introduce the concept of slots is here in the directives section, I think they're a bit more fundamental and could go in the quickstart guide. However I think leave that for now and come back to it after the 5th